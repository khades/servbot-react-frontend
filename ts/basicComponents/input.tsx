import * as React from "react";
import ControlGroup from "./control-group";

export interface IInputProps {
    className?: string;
    errors?: string | string[];
    id: string;
    label: string | string[];
    value: string;
    setValue: (value: string, id?: string) => void;
    placeholder?: string;
}

export default class Input extends React.PureComponent<IInputProps, {}> {

    public render() {
        return (
            <ControlGroup
                className={this.props.className}
                id={this.props.id}
                errors={this.props.errors}
                label={this.props.label}
            >
                <input
                    id={this.props.id}
                    onChange={this.setValue}
                    defaultValue={this.props.value}
                    placeholder={this.props.placeholder}
                />
            </ControlGroup>
        );
    }

    private setValue = (event: React.FormEvent<HTMLInputElement>) => {
        this.props.setValue(event.currentTarget.value, this.props.id);
    }

}
