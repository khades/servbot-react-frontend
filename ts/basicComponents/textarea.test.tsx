import * as React from "react";
import { create } from "react-test-renderer";
import { IInputProps } from "./input";
import Textarea from "./textarea";

describe("Textarea", () => {
    it("value change should work", () => {
        let inputValue = "";
        const props: IInputProps = {
            id: "ID",
            getLabel: () => "LABEL",
            getValue: () => inputValue,
            setValue: (value: string) => inputValue = value,
        };

        const input = create(
            <Textarea {...props} />,
        );

        const inputInstance = input.root.find((el) => el.type === "textarea");
        inputInstance.props.onChange({ currentTarget: { value: "2" } });
        input.update(
            <Textarea {...props} />,
        );
        expect(inputValue).toBe("2");
    });
    it("inital value should be properly set", () => {
        const props: IInputProps = {
            id: "ID",
            getLabel: () => "LABEL",
            getValue: () => "testValue",
            setValue: (value: string) => value,
        };

        const input = create(
            <Textarea {...props} />,
        );

        const inputInstance = input.root.find((el) => el.type === "textarea");
        expect(inputInstance.props.defaultValue).toBe("testValue");
    });
});
