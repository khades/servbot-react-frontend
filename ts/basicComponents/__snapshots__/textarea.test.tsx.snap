// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea inital value should be properly set 1`] = `
<div
  className="control-group"
>
  <label
    className="control-group__label"
    htmlFor="ID"
  >
    LABEL
  </label>
  <textarea
    id="ID"
    onChange={[Function]}
    value="testValue"
  />
</div>
`;
